File: order-service/src/models/order.rs
use bigdecimal::BigDecimal;
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow)]
pub struct Order {
    pub id: Uuid,
    pub user_id: Uuid,
    pub status: String,
    pub total: BigDecimal,
    pub shipping_address: String,
    pub created_at: NaiveDateTime,
}

#[derive(Debug, Deserialize)]
pub struct CreateOrderRequest {
    pub shipping_address: String,
    pub items: Vec<CreateOrderItemRequest>,
}

#[derive(Debug, Deserialize)]
pub struct CreateOrderItemRequest {
    pub product_id: Uuid,
    pub quantity: i32,
    pub unit_price: BigDecimal,
}

#[derive(Debug, Deserialize)]
pub struct UpdateStatusRequest {
    pub status: String,
}

#[derive(Debug, Serialize)]
pub struct OrderResponse {
    pub id: Uuid,
    pub user_id: Uuid,
    pub status: String,
    pub total: BigDecimal,
    pub shipping_address: String,
    pub created_at: NaiveDateTime,
    pub items: Vec<super::order_item::OrderItemResponse>,
}

File: frontend/src/types/order.ts
export interface OrderItem {
  id: string;
  orderId: string;
  productId: string;
  productName?: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
}

export interface Order {
  id: string;
  userId: string;
  status: string; // e.g., PENDING, PAID, CANCELLED
  total: number;
  shippingAddress: string;
  createdAt: string;
  items: OrderItem[];
}

export interface CreateOrderPayload {
  shipping_address: string;
  items: {
    product_id: string;
    quantity: number;
    unit_price: number;
  }[];
}
